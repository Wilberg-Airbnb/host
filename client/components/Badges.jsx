import React from 'react';
import {List,Grid, Segment} from 'semantic-ui-react';
import axios from 'axios';
import { VerifiedUserRounded, StarRounded, CheckCircleRounded } from '@material-ui/icons';

const iconStyle = {
  color: '#FF385C',
  verticalAlign: 'bottom'
};

const contentStyle = {
  marginLeft: '6px',
  display: 'inline'
}

const itemStyle = {
  padding: '0px 12px 0px 12px',
  margin: '0px'
}

class Badges extends React.Component {

  constructor(props) {
  super(props);
  this.state = {
    reviews: null
    }
  }

  componentDidMount(){

    const url = 'http://52.14.214.44:8080/api/reviews';
    // const url = 'http://localhost:2000/api/reviews';
    const options = {
      params: { array: JSON.stringify(this.props.listingIds)},
  
    };

    axios.get(url, options)
    .then ( (res) => {
      const reviews = res.data;
      // console.log(reviews);
      this.setState({reviews});
    })
    .then( (err) => {
      console.log(err);
    });
  }

  render(){

 
    return(

    <List stackable horizontal >
      <List.Item style={itemStyle}>
        <StarRounded style={iconStyle}/>
        <List.Content style={contentStyle}>
          {this.state.reviews} Reviews
        </List.Content>
      </List.Item>
      {(this.props.verification === "Identity verified") ?
      <List.Item style={itemStyle}>
        <VerifiedUserRounded style={iconStyle}/>
        <List.Content style={contentStyle}>
          Identity verified
        </List.Content>
      </List.Item>
        : 
        <List.Item style={itemStyle}>
        <CheckCircleRounded style={iconStyle}/>
        <List.Content style={contentStyle}>
          Verified
        </List.Content>
      </List.Item>}
        {this.props.superhost ?
          <List.Item style={itemStyle}>
            <i className="material-icons" style={iconStyle}>military_tech</i>
            <List.Content style={contentStyle}>
              Superhost
            </List.Content>
          </List.Item>: null}
    </List>)
  }
}

export default Badges;