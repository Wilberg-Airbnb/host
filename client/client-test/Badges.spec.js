import React from 'react';
import Badges from '../components/Badges.jsx';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });
 
describe('Renders Components', () => {

  let component;
  let props = {
    verification: 'Identity verified',
    superhost: true,
    listingIds: ['4','5']
    };
  beforeEach( () => {
    component = shallow(<Badges verifcation={props.verification} superhost={props.superhost} listingIds={props.listingIds}/>)
  });

  test("should render initial layout", () => {
    const component = shallow(<Badges verifcation={props.verification} superhost={props.superhost} listingIds={props.listingIds}/>);
    expect(component.getElements()).toMatchSnapshot();
  });


});


