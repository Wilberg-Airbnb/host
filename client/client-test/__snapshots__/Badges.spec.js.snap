// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Components should render initial layout 1`] = `
Array [
  <List
    horizontal={true}
    relaxed={true}
  >
    <ListItem>
      <Memo(StarRoundedIcon)
        style={
          Object {
            "color": "#FF385C",
            "verticalAlign": "bottom",
          }
        }
      />
      <ListContent
        style={
          Object {
            "display": "inline",
            "marginLeft": "6px",
          }
        }
      >
         Reviews
      </ListContent>
    </ListItem>
    <ListItem>
      <Memo(CheckCircleRoundedIcon)
        style={
          Object {
            "color": "#FF385C",
            "verticalAlign": "bottom",
          }
        }
      />
      <ListContent
        style={
          Object {
            "display": "inline",
            "marginLeft": "6px",
          }
        }
      >
        Verified
      </ListContent>
    </ListItem>
    <ListItem>
      <ListContent
        style={
          Object {
            "display": "inline",
            "marginLeft": "6px",
          }
        }
      >
        <i
          className="material-icons"
          style={
            Object {
              "color": "#FF385C",
              "verticalAlign": "bottom",
            }
          }
        >
          military_tech
        </i>
        Superhost
      </ListContent>
    </ListItem>
  </List>,
]
`;
